analytics/src/report-generator.js

```diff
@@ -0,0 +1,70 @@
+const fs = require("fs");
+const path = require("path");
+class ReportGenerator {
+  constructor(outpath) {
+    this.outpath = outpath;
+  }
+  generate(filename, filedata) {
+    const data = this.jsonToCSV(filedata);
+
+    if (this.outputFolderExists()) {
+      fs.writeFileSync(path.join(this.outpath, filename), data);
+    } else {
+      fs.mkdirSync(this.outpath);
+      if (this.outputFolderExists()) {
+        fs.writeFileSync(path.join(this.outpath, filename), data);
+      }
+    }
+    return data;
+  }
+
+  previousDay(date) {
+    return date.getDate() - 1;
+  }
+
+  getReportDateString() {
+    const date = new Date();
+    const yesterday = this.previousDay(date);
+    date.setDate(yesterday);
+
+    const year = String(date.getFullYear());
+    const month = String(date.getMonth() + 1).padStart(2, "0");
+    const dayOfMonth = String(date.getDate()).padStart(2, "0");
+
+    return `${year}-${month}-${dayOfMonth}`;
+  }
+
+  jsonToCSV(jsonArrayIn) {
+    const outLines = [];
+    const dateString = this.getReportDateString();
+
+    outLines.push(`"Article Title","Review content","Date"`);
+    jsonArrayIn.map((review, j) => {
+      let reviewsExist = review.reviews.length > 0;
+      if (reviewsExist && review.title.includes("REVIEW")) {
```

Comment:
Assumes existence of review.title, and assumes it is a string?

Reply:
doesn't assume its existence, but now that we check the existence of review, it will be come back as undefined if doesn't exist rather than crashing. Would fail the if criteria anyway, so I think the if(review) fixes this one too